### ISSUE TITLE
Numpy version incompatibility with Aer 0.4.0

### ISSUE THREAD

**Issue by woodsp-ibm** [Qiskit/qiskit-aer#592]
-  **Qiskit Aer version**: 0.4
- **Python version**: 3.6+
- **Operating system**: Linux

When Terra tries to load Aer provider, with numpy 1.15.4, which was default on our Aqua build, and conforms to > 1.13 we get this

```
line 440, in <module>
    from .cy.spmath import (zcsr_transpose, zcsr_adjoint, zcsr_mult)
  File "stringsource", line 105, in init qiskit.providers.aer.pulse.qutip_lite.cy.spmath
AttributeError: type object 'qiskit.providers.aer.pulse.qutip_lite.cy.spmath.ar' has no attribute '__reduce_cython__'
```
We have tried locally and get similar error with 1.15.4. (I was running 1.16.3 which seems ok, and latest 1.18.1 seemed fine too)

**Comment by nonhermitian** [comment]
Just for the record, this is a bug in Cython 0.27, not a NumPy issue. It has been fixed in 0.28 and later. 

https://github.com/cython/cython/issues/1953

**Comment by yulunwang** [comment]
> Just for the record, this is a bug in Cython 0.27, not a NumPy issue. It has been fixed in 0.28 and later.
> 
> [cython/cython#1953](https://github.com/cython/cython/issues/1953)

Hi @nonhermitian, I have the same error here but my Cython is 0.29.2. Do you have any idea? Thx

**Comment by nonhermitian** [comment]
Yeah. They said it fixed but it is not. Numpy needs to be updated.
